server/index.js

```diff
@@ -161,6 +158,33 @@ app.post('/api/cart/:productId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.post('/api/orders', (req, res, next) => {
+  const { name, creditCard, shippingAddress } = req.body;
+  if (!req.session.cartId) {
+    return res.status(400).json({ error: 'must have cart id' });
+  } else if (!name || !creditCard || !shippingAddress) {
+    return res.status(400).json({ error: 'name creditCard and shippingAddress are required fields' });
+  } else {
+    const sql = `
+    insert into "orders" ("cartId", "name", "creditCard", "shippingAddress")
+    values ($1, $2, $3, $4)
+    returning "orderId", "createdAt", "name", "creditCard", "shippingAddress";
+    `;
+    const values = [`${req.session.cartId}`, `${name}`, `${creditCard}`, `${shippingAddress}`];
+
+    db.query(sql, values)
+      .then(response => {
+
+        if (!response.rows) {
+          next(response);
+        } else {
```

Comment:
You can remove this conditional. `result.rows` will always be an array and arrays are truthy.